lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -343,7 +343,11 @@ defmodule Mix.Tasks.Xref do
   ## Print callers
 
   defp print_calls(calls) do
-    Enum.map(calls, &IO.write(format_call(&1)))
```

Comment:
I think this should be:

    Enum.map(calls, &Mix.shell().info(format_call(&1)))

and you will have to change the tests to accumulate the result of many calls and merge them. See `receive_until_no_messages` in xref_test.

Reply:
I stumbled upon a problem there, since `locations` in `format_call` can have multiple call sites for the given  MFA, `format_call` itself returns a list, which would then go straight into `Mix.shell().info`, therefore printing multiple lines without a linebreak (if we change `format_call` / `format_unreachable` to not add newlines`). That was the original idea behind the `flat_map` here. We could drop the newlines from the `format_*` and join with a newline maybe?

Reply:
Alternatively, we could not join the `flat_map` results at all and just call `Mix.shell().info` for every line, if that makes more sense

Reply:
@jfornoff we can also change `format_call` to `print_call` and write it like this:

```
defp print_calls(calls) do
  Enum.each(calls, &print_call/1)
  calls
end

defp print_call({{module, func, arity}, locations}) do
  shell = Mix.shell()

  for {file, line} <- locations do
    shell.info([
      Exception.format_file_line(file, line, " "),
      Exception.format_mfa(module, func, arity)
    ])
  end
end
```

Thoughts?

Reply:
Absolutely, sounds good. I'll add that!